app/src/main/java/com/kickstarter/libs/utils/ApplicationLifecycleUtil.java

```diff
@@ -44,24 +49,34 @@ public void onActivityStarted(final @NonNull Activity activity) {
 
   @Override
   public void onActivityResumed(final @NonNull Activity activity) {
-    if(this.isInBackground){
-
+    if(this.isInBackground) {
       // Facebook: logs 'install' and 'app activate' App Events.
       AppEventsLogger.activateApp(activity.getApplication());
 
+      //to get the current config for segment feature when app resume from background
+      this.config.observable().subscribe(c -> {
+        this.currentConfig = c;
+      });
+
       // Refresh the config file
       this.client.config()
-        .compose(Transformers.pipeApiErrorsTo(this::handleConfigApiError))
-        .compose(Transformers.neverError())
-        .subscribe(c -> this.config.config(c));
+              .compose(Transformers.pipeApiErrorsTo(this::handleConfigApiError))
+              .compose(Transformers.neverError())
+              .subscribe(c -> {
+                if (Objects.requireNonNull(c.features()).containsKey(ConfigFeatureName.SEGMENT_ENABLED.getConfigFeatureName())) {
```

Comment:
There is no way for us to override the feature flags without having to touch any actual production code? what would happen when we have more feature flags?

Reply:
the issue that code called every time app resumed  so if the app in the background and come to foreground the config is ovverrided

Reply:
Done ✅